core/block/editor/template/collection.go

```diff
@@ -9,68 +11,137 @@ import (
+		bundle.RelationKeyName,
+		bundle.RelationKeyCreatedDate,
+		bundle.RelationKeyCreator,
+		bundle.RelationKeyLastModifiedDate,
+		bundle.RelationKeyLastModifiedBy,
+		bundle.RelationKeyLastOpenedDate,
+		bundle.RelationKeyBacklinks,
 	}
-	viewRelations := []*model.BlockContentDataviewRelation{
-		{
-			Key:       bundle.RelationKeyName.String(),
-			IsVisible: true,
-		},
+
+	defaultCollectionRelations = []domain.RelationKey{
+		bundle.RelationKeyName,
+		bundle.RelationKeyType,
+		bundle.RelationKeyCreatedDate,
+		bundle.RelationKeyCreator,
+		bundle.RelationKeyLastModifiedDate,
+		bundle.RelationKeyLastModifiedBy,
+		bundle.RelationKeyLastOpenedDate,
+		bundle.RelationKeyBacklinks,
+		bundle.RelationKeyTag,
+		bundle.RelationKeyDescription,
 	}
-	for _, relKey := range DefaultDataviewRelations {
-		if pbtypes.HasRelationLink(relations, relKey.String()) {
-			continue
+
+	defaultVisibleRelations = []domain.RelationKey{
+		bundle.RelationKeyName,
+		bundle.RelationKeyType,
+	}
+)
+
+func MakeDataviewContent(isCollection bool, ot *model.ObjectType, relLinks []*model.RelationLink) *model.BlockContentOfDataview {
+	var (
+		defaultRelations = defaultCollectionRelations
+		visibleRelations = defaultVisibleRelations
+		sorts            = DefaultLastModifiedDateSort()
+	)
+
+	if isCollection {
+		sorts = defaultNameSort()
+	} else if relLinks != nil {
+		for _, relLink := range relLinks {
+			visibleRelations = append(visibleRelations, domain.RelationKey(relLink.Key))
 		}
-		rel := bundle.MustGetRelation(relKey)
-		relations = append(relations, &model.RelationLink{
-			Format: rel.Format,
-			Key:    rel.Key,
-		})
-		viewRelations = append(viewRelations, &model.BlockContentDataviewRelation{Key: rel.Key, IsVisible: false})
+	} else if ot != nil {
+		defaultRelations = defaultDataviewRelations
+		relLinks = ot.RelationLinks
+	} else {
+		defaultRelations = defaultDataviewRelations
 	}
 
-	blockContent := &model.BlockContentOfDataview{
+	relationLinks, viewRelations := generateRelationLists(defaultRelations, relLinks, visibleRelations)
+
+	return &model.BlockContentOfDataview{
 		Dataview: &model.BlockContentDataview{
-			IsCollection:  true,
-			RelationLinks: relations,
+			IsCollection:  isCollection,
+			RelationLinks: relationLinks,
 			Views: []*model.BlockContentDataviewView{
 				{
-					Id:   bson.NewObjectId().Hex(),
-					Type: model.BlockContentDataviewView_Table,
-					Name: "All",
-					Sorts: []*model.BlockContentDataviewSort{
-						{
-							RelationKey: "name",
-							Type:        model.BlockContentDataviewSort_Asc,
-						},
-					},
+					Id:        bson.NewObjectId().Hex(),
+					Type:      model.BlockContentDataviewView_Table,
+					Name:      defaultViewName,
+					Sorts:     sorts,
 					Filters:   nil,
 					Relations: viewRelations,
 				},
 			},
 		},
 	}
-	return blockContent
 }
 
-var DefaultDataviewRelations = make([]domain.RelationKey, 0, len(bundle.RequiredInternalRelations))
+func generateRelationLists(
```

Comment:
Personally, I would rename this function, because it doesn't generate anything 🧐 Maybe `retrieveViewRelations`?